### PR TITLE
Bump to k8s 1.16.0

### PR DESCRIPTION
**Pull request by adambkaplan**
1. Switch to go mod
2. Remove glide
3. Update deps to k8s 1.16.0 and latest openshift api, client-go, library-go

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,7 +27,9 @@ $(call build-image,ocp-cluster-openshift-controller-manager-operator,$(IMAGE_REG
 # $5 - output
 # It will generate targets {update,verify}-bindata-$(1) logically grouping them in unsuffixed versions of these targets
 # and also hooked into {update,verify}-generated for broader integration.
-$(call add-bindata,v3.11.0,./bindata/v3.11.0/...,bindata,v311_00_assets,pkg/operator/v311_00_assets/bindata.go)
+
+# TODO: turn this back on once compatible with go mod
+# $(call add-bindata,v3.11.0,./bindata/v3.11.0/...,bindata,v311_00_assets,pkg/operator/v311_00_assets/bindata.go)
```
Comment (adambkaplan) [inline Makefile]: Need to restore add-bindata

File: Makefile [hunk 2]
```diff
@@ -59,4 +60,9 @@ verify-codegen-crds: crd-schema-gen
 verify-codegen: verify-codegen-crds
 .PHONY: verify-codegen
 
-verify: verify-codegen
+# TODO: fix crd-schema-gen to work with go modules
+# verify: verify-codegen
```
Comment (adambkaplan) [inline Makefile]: Need to restore verify-codegen

File: Makefile [hunk 3]
```diff
@@ -59,4 +60,9 @@ verify-codegen-crds: crd-schema-gen
...
+
+verify-deps:
+	echo "TODO: re-enable verify-deps for go mod"
+.PHONY: verify-deps
```
Comment (adambkaplan) [inline Makefile]: Need to restore verify-deps

**Comment by soltysh** [comment]
> @deads2k @mfojtik @soltysh do we have any means of restoring the verify coverage with go-mod?

we do but not that high priority, if you're interested we're open :smiley:

**Comment by soltysh** [comment]
@adambkaplan honestly this PR is hard to review, have a look at how we try to structure bumps + switch to go.mod https://github.com/openshift/cluster-kube-controller-manager-operator/pull/290 and https://github.com/openshift/cluster-kube-controller-manager-operator/pull/292 it's fine to have it done within single PR but please make the commits easy to follow.

**Comment by adambkaplan** [comment]
@soltysh as with ocm, I think I'm going to split the switch and k8s.io bump in two PRs.

**Pull request closed without merging**